node/src/components/consensus/highway_core/state.rs

```diff
@@ -315,14 +322,13 @@ impl<C: Context> State<C> {
         unit: &C::Hash,
         v_ids: I,
     ) -> bool {
-        self.incomplete_endorsements
-            .get(unit)
-            .map(|v| {
-                v_ids
-                    .into_iter()
-                    .all(|v_id| v.get(v_id.0 as usize).is_some())
-            })
-            .unwrap_or(false)
+        if let Some(sigs) = self.incomplete_endorsements.get(unit) {
+            v_ids.into_iter().all(|v_id| sigs.contains_key(v_id))
+        } else if let Some(sigs) = self.endorsements.get(unit) {
+            v_ids.into_iter().all(|v_id| sigs[*v_id].is_some())
```

Comment:
We do `state.is_endorsed` check first, before calling `has_all_endorsements`.

Reply:
I see; but this is a cheap check and I find it more intuitive to do that inside this method. Do you want me to remove it anyway?

Reply:
Maybe switch places, since it should be cheaper than the first one.

Reply:
And then, remove the other call to `state.is_endorsed`. We don't want to repeat ourselves.